torchaudio/datasets/voxceleb1.py

```diff
@@ -0,0 +1,206 @@
+            if not download:
+                raise RuntimeError(
+                    f"Dataset not found at {self._path}. Please set `download=True` to download the dataset."
+                )
+            _download_extract_wavs(root)
+
+    def __getitem__(self, n: int):
+        raise NotImplementedError
+
+    def __len__(self) -> int:
+        raise NotImplementedError
+
+
+class VoxCeleb1Identification(VoxCeleb1):
+    """Create *VoxCeleb1* [:footcite:`nagrani2017voxceleb`] Dataset for speaker identification task.
+    Each data sample contains the waveform, correcponding speaker class label, sample rate, and the file name.
+
+    Args:
+        root (str or Path): Path to the directory where the dataset is found or downloaded.
+        subset (str, optional): Subset of the dataset to use. Options: ["train", "dev", "test"]. (Default: ``"train"``)
+        mate_url (str, optional): The url of meta file that contains the list of subset labels and file paths.
+        download (bool, optional):
+            Whether to download the dataset if it is not found at root path. (Default: ``False``).
+    """
+
+    def __init__(
+        self, root: Union[str, Path], subset: str = "train", meta_url: str = _IDEN_SPLIT_URL, download: bool = False
+    ) -> None:
+        super().__init__(root, download)
+        assert subset in ["train", "dev", "test"], "`subset` must be one of ['train', 'dev', 'test']"
+        # download the iden_split.txt to get the train, dev, test lists.
+        meta_list_path = os.path.join(root, os.path.basename(meta_url))
+        if not os.path.exists(meta_list_path):
+            download_url_to_file(meta_url, meta_list_path)
+        self._flist = _get_flist(self._path, meta_list_path, subset)
+
+    def __getitem__(self, n: int) -> Tuple[Tensor, int, int, str]:
+        """Load the n-th sample from the dataset.
+
+        Args:
+            n (int): The index of the sample to be loaded
+
+        Returns:
+            (Tensor, int, int, str):
+            ``(waveform, speaker_id, sample_rate, file_id)``
+        """
+        file_path = self._flist[n]
+        file_id = _get_file_id(file_path, self._ext_audio)
+        speaker_id = file_id.split("-")[0]
+        speaker_id = int(speaker_id[3:])
+        waveform, sample_rate = torchaudio.load(os.path.join(self._path, file_path))
+        return (waveform, speaker_id, sample_rate, file_id)
+
+    def __len__(self) -> int:
+        return len(self._flist)
+
+
+class VoxCeleb1Verification(VoxCeleb1):
+    """Create *VoxCeleb1* [:footcite:`nagrani2017voxceleb`] Dataset for speaker verification task.
+    Each data sample contains a pair of waveforms, the label indicating if they are from the same speaker,
+    sample rate, and the file names.
+
+    Args:
+        root (str or Path): Path to the directory where the dataset is found or downloaded.
+        subset (str, optional): Subset of the dataset to use. Options: ["train", "dev", "test"]. (Default: ``"train"``)
```

Comment:
`subset` is unused in the current implementation here. In [s3prl],(https://github.com/s3prl/s3prl/blob/master/s3prl/downstream/sv_voxceleb1/dataset.py#L30) there is support for train and test splits, and they have different implementations for getitem

Reply:
Thanks. This line should be removed.
The `SpeakerVerifi_train` dataset in s3prl can be replaced by our `VoxCeleb1Identification` dataset. Given one utterance as the input, the label is the class of the speaker. `SpeakerVerifi_test` is for a different task, given two utterances, the label indicates whether they are from the same speaker or not, which can be replaced by our `VoxCeleb1Verification` dataset.

Reply:
this line still needs to be removed